x-pack/plugins/apm/public/components/alerting/TransactionDurationAnomalyAlertTrigger/index.tsx

```diff
@@ -59,7 +59,7 @@ export function TransactionDurationAnomalyAlertTrigger(props: Props) {
     [TRANSACTION_PAGE_LOAD, TRANSACTION_REQUEST].includes(transactionType)
   );
 
-  if (!supportedTransactionTypes.length || !serviceName) {
+  if (serviceName && !supportedTransactionTypes.length) {
     return null;
   }
```

Comment:
Please update the comment `// 'page-load' for RUM, 'request' otherwise`. Perhaps make the code more clear if you can to indicate that it might also be `undefined`.